packages/node-app/src/controllers/giveaway.controller.ts

```diff
@@ -41,11 +32,30 @@ export const listGiveaways = async (req: Request, res: Response) => {
       )
       .lean();
 
-    giveaways = giveaways.map((giveaway) => ({
-      ...omit(giveaway, ['participants']),
-      winners: giveawayWinners(giveaway),
-      stats: giveawayStats(giveaway),
-    }));
+    const active = req.query.active === 'true' ? true : false;
+    const archived = req.query.active === 'false' ? true : false;
+
+    const activeGiveaways = getActiveGiveaways(giveaways, req.user.role);
+
+    giveaways = active
+      ? activeGiveaways.map((giveaway) => ({
+          ...omit(giveaway, ['participants']),
+          winners: giveawayWinners(giveaway),
+          stats: giveawayStats(giveaway),
+        }))
+      : archived
+      ? giveaways
+          .filter((g) => !activeGiveaways.includes(g))
+          .map((giveaway) => ({
+            ...omit(giveaway, ['participants']),
+            winners: giveawayWinners(giveaway),
+            stats: giveawayStats(giveaway),
+          }))
+      : giveaways.map((giveaway) => ({
+          ...omit(giveaway, ['participants']),
+          winners: giveawayWinners(giveaway),
+          stats: giveawayStats(giveaway),
+        }));
```

Comment:
Maybe we could refractor this part to be more clear, and also to avoid getting the active giveaways if not needed

```suggestion

   if (req.query.active) {
      const active = req.query.active === 'true'
      const activeGiveaways = getActiveGiveaways(giveaways, req.user.role);
      
      if (active) {
         giveaways = activeGiveaways
      } else {
         giveaways = giveaways.filter((g) => !activeGiveaways.includes(g))
      }
   }

   giveaways = giveaways.map((giveaway) => ({
      ...omit(giveaway, ['participants']),
      winners: giveawayWinners(giveaway),
      stats: giveawayStats(giveaway),
    }));